### DISCUSSION TITLE
How to add a List<Note> or a Scale to PatternBuilder?

### DISCUSSION THREAD

**Discussion by seanjmiller** [melanchall/drywetmidi#215] [General]
Fantastic project!
I don't see how I can add a List<Note> to a PatternBuilder or even a Scale to the PatternBuilder.
The examples show notes known ahead of time, but what if I want to supply a List<Note> that may have 7, 12, ... notes?

**Comment by melanchall** [comment]
You can just add notes in a loop.

```csharp
foreach (var note in notesList)
{
    patternBuilder.Note(note);
}
```

**Comment by seanjmiller** [comment]
Ok thanks. What was throwing me off was that you can't use foreach in the Fluent callout and that a .Build() needs to be called on the added Pattern.

`
        var pattern1 = new PatternBuilder();

        foreach (var note in notes)
        {
            pattern1.Note(note);
        }

        Pattern pattern = new PatternBuilder()
            .SetNoteLength(MusicalTimeSpan.Quarter)

            .Pattern(pattern1.Build())

            // Get pattern
            .Build();
`

**Reply by melanchall** [reply]
Why so complicated?

```csharp
var patternBuilder = new PatternBuilder().SetNoteLength(MusicalTimeSpan.Quarter);

foreach (var note in notes)
{
    patternBuilder.Note(note);
}

Pattern pattern =  patternBuilder.Build();
```

**Reply by melanchall** [reply]
@seanjmiller Any news?

**Comment by seanjmiller** [comment]
Yes this works. It just took me with your help and poking through the documentation to do what I need to do with your libraries. The fluent approach just through me for a loop. Thanks for the help. I didn't see a way to mark your response as the answer or close this issue out.
